javascript/ql/src/Expressions/MissingReturnValueGetsUsed.ql

```diff
@@ -0,0 +1,88 @@
+/**
+ * @name Missing Return Value Gets Used
+ * @description A call to a function that doesn't return a value is used in a context where the
+ *              value of the call actually matters.
+ * @kind problem
+ * @problem.severity warning
+ * @id js/missing-return-value-gets-used
+ * @precision very-high
```

Comment:
A quick experiment suggests that this query is quite noisy in practice: https://lgtm.com/query/5744005583484976286

Perhaps you could take a quick look at the results to get some ideas for how to sharpen the query to make the results more interesting (remembering Oege's dictum that "a false positive is any result that the developer does not want to fix").

Reply:
Will do. Do you have any recommendations for how to approach certain cases? For instance, I think one common thing I've seen is people relying on the falsiness of `undefined` to allow them to not have to write a return statement, but they really *do* care about the result of the function call, eg as a test.

I feel like that might be a case where some more type-like information is useful because that at least tells us more about the use context and whether or not `undefined` might be acceptable in it.

Reply:
I'm afraid I don't have any concrete suggestions. I think you'd probably want to sharpen your criteria for what constitutes an unproblematic use of an undefined return value, or perhaps turn it round and look for uses that are most likely problematic, such as accessing a property on it, or using it in arithmetic or string operations.

Reply:
Sounds good. I'll look at modeling relevance and see what I can write up.